backend/hdf5/h5x/H5DataType.cpp

```diff
@@ -281,6 +281,12 @@ data_type_from_h5(H5T_class_t vclass, size_t vsize, H5T_sign_t vsign)
     return DataType::Nothing;
 }
 
+void check_bool_enum(h5x::DataType type) {
+    assert(type.member_count() == 2);
```

Comment:
`assert` statements can be turned into NOOPs with the right define (NDEBUG), which is not a good idea. Throwing an exception, or returning `DataType::Nothing`.